### PR TITLE
Add the 'exciting' package.

### PR DESCRIPTION
**Pull request by downloadico**
Version 14 (latest available) is defined.
An as-of-yet unpublished patch (dfgather.patch) from the developers is also
included.

### CODE REVIEW

File: var/spack/repos/builtin/packages/exciting/package.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    ' '.join(['FCFLAGS = @FCFLAGS@', '-cpp',
+                              self.compiler.openmp_flag]),
+                    'src/libXC/src/Makefile.in')
+        if '+mkl' in spec:
+            if '%intel' in spec:
+                opts['LIB_LPK'] = '-mkl=parallel'
+            opts['INC_MKL'] = spec['mkl'].headers.include_flags
+            opts['LIB_MKL'] = spec['mkl'].libs.ld_flags
+        else:
+            opts['LIB_LPK'] = ' '.join([spec['lapack'].libs.ld_flags,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/exciting/package.py:70]: I think you're missing a dependency on lapack

File: var/spack/repos/builtin/packages/exciting/package.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        opts['LIB_FFT'] = 'fftlib.a'
+        opts['LIB_BZINT'] = 'libbzint.a'
+        opts['LIBS'] = '$(LIB_ARP) $(LIB_LPK) $(LIB_FFT) $(LIB_BZINT)'
+        with open('build/make.inc', 'a') as inc:
+            for key in opts:
+                inc.write('{0} = {1}\n'.format(key, opts[key]))
+
+    def install(self, spec, prefix):
+        cd = which('cd')
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/exciting/package.py]: You'll want to do this in the `build` method, not in `install`. The build method already runs `make` for you, so you can probably remove this.

File: var/spack/repos/builtin/packages/exciting/package.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        cd('build')
+        install_tree('bin', prefix.bin)
+        cd('..')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/exciting/package.py]:
```suggestion
        with working_dir('build'):
            install_tree('bin', prefix.bin)
```

File: var/spack/repos/builtin/packages/exciting/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/exciting/package.py:70]: You're right!  I added it as a dependency.

File: var/spack/repos/builtin/packages/exciting/package.py (hunk 2 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/exciting/package.py]: right!  fixed now.

File: var/spack/repos/builtin/packages/exciting/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/exciting/package.py:70]: Did you forget to push the commit?

Comment (downloadico) [reply var/spack/repos/builtin/packages/exciting/package.py:70]: I /think/ I did.  'git push' says everything is up to date.

Comment (downloadico) [reply var/spack/repos/builtin/packages/exciting/package.py:70]: nevermind.... I just pushed actually requiring lapack!

File: var/spack/repos/builtin/packages/exciting/package.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    version('14', sha256='a7feaffdc23881d6c0737d2f79f94d9bf073e85ea358a57196d7f7618a0a3eff')
+
+    # as-of-yet unpublished fix to version 14
+    patch('dfgather.patch', when='@14', working_dir='src/src_xs', level=0)
+    variant('mpi', default=False, description='Use MPI')
+    variant('mkl', default=False, description='Use MKL')
+    variant('omp', default=True, description='Use OpenMP')
+    variant('scalapack', default=False, description='Use ScaLAPACK')
+    depends_on('blas+lp64')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/exciting/package.py]: I don't think virtual dependencies support variants at the moment.

Comment (downloadico) [reply var/spack/repos/builtin/packages/exciting/package.py]: You're right!  I removed it.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**